json_serializable/lib/src/type_helpers/enum_helper.dart

```diff
@@ -46,8 +53,19 @@ class EnumHelper extends TypeHelper {
 
     final functionName =
         context.nullable ? r'_$enumDecodeNullable' : r'_$enumDecode';
-    return '$functionName(${_constMapName(targetType)}, '
-        '$expression)';
+
+    final args = [
+      _constMapName(targetType),
+      expression,
+    ];
+
+    final jsonKey = jsonKeyForField(context.fieldElement, context.config);
+    // TODO(kevmoo): use collection expressions once min-SDK is >= 2.3.0
```

Comment:
Why not bump now?

Reply:
Trying to minimize churn in this CL. Bumping just one package means travis doubles jobs. Then I'd have to go bump the other packages, etc etc etc